t/ust.t

```diff
@@ -0,0 +1,366 @@
+                        'vsize'   => 369,
+                        'version' => 2,
+                        'size'    => 369,
+                        'hash'    => 'abd322cee71dbd9adbd8b200df4c69a7870251f73ec6e04a3eb5e9aa2b065bdf'
+                    }
+
+                ],
+                'bits' => '1d00ffff',
+                'time' => 1582166123
+            };
+            return Future->done($getblock);
+        },
+        list_by_addresses => sub {
+            my ($self, $address) = @_;
+            my $result;
+            if ($address eq $from) {
+                $result = [{
+                        txids   => [$hash],
+                        address => $from
+                    }];
+            }
+
+            else { $result = []; }
+
+            return Future->done($result);
+        },
+        get_transaction => sub {
+            my ($self, @params) = @_;
+            my $omni_gettransaction = {
+                blocktime        => 1582166123,
+                confirmations    => 1,
+                blockhash        => $blockhash,
+                block            => 1666276,
+                type             => 'Simple Send',
+                amount           => '21.00000000',
+                divisible        => 1,
+                fee              => '0.00000256',
+                positioninblock  => 352,
+                sendingaddress   => $from,
+                ismine           => 1,
+                version          => 0,
+                txid             => $hash,
+                type_int         => 0,
+                referenceaddress => $to,
+                propertyid       => 2147484941,
+                valid            => 1
+            };
+
+            return Future->done($omni_gettransaction);
+        });
+
+    my $subscription_source = $subscription_client->source;
+    $subscription_source->each(
+        sub {
+            my $transaction = shift;
+            is $transaction->{currency}, 'UST',  "Currency code: $transaction->{currency} is correct";
+            is $transaction->{amount},   "21",   "Amount is correct";
+            is $transaction->{type},     'sent', "Transaction Type: $transaction->{type} is correct";
+            is_deeply $transaction, $expected_transaction, "$currency_code Omni-Send Transaction is emitted correctly.";
+            $subscription_source->finish();
+        });
+
+    my $emitted_transaction = $subscription_client->hashblock($blockhash)->get;
+
```

Comment:
you should call here `$subscription_source->get;` to make sure the future has been ended